airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlInitialLoadHandler.java

```diff
@@ -92,6 +96,15 @@ public List<AutoCloseableIterator<AirbyteMessage>> getIncrementalIterators(
             .map(CommonField::getName)
             .filter(CatalogHelpers.getTopLevelFieldNames(airbyteStream)::contains)
             .collect(Collectors.toList());
+
+        // This is to handle the case if the user de-selects the PK column
+        // Necessary to query the data via pk but won't be added to the final record
+        primaryKeys.forEach(pk -> {
```

Comment:
@rodireich FYI: This is how the PK de-selection is handled.

Reply:
And this falls back to regular initial sync (non pk)?

Reply:
Oh no, so if a customer de-selects the PK column, we still add it back to the `selectedDatabaseFields` so we can query it, but since it's not enabled in the catalog, platform will omit it for us.

Reply:
got it.
And that means we're also going to send this column down to destination